include/crocoddyl/multibody/costs/cost-sum.hpp

```diff
@@ -143,12 +143,45 @@ struct CostDataSum {
   Eigen::MatrixXd get_Lxu() const { return Lxu; }
   Eigen::MatrixXd get_Luu() const { return Luu; }
 
-  void set_r(Eigen::VectorXd _r) { r = _r; }
-  void set_Lx(Eigen::VectorXd _Lx) { Lx = _Lx; }
-  void set_Lu(Eigen::VectorXd _Lu) { Lu = _Lu; }
-  void set_Lxx(Eigen::MatrixXd _Lxx) { Lxx = _Lxx; }
-  void set_Lxu(Eigen::MatrixXd _Lxu) { Lxu = _Lxu; }
-  void set_Luu(Eigen::MatrixXd _Luu) { Luu = _Luu; }
+  void set_r(const Eigen::VectorXd& _r) {
+    if (r.size() != _r.size()) {
+      throw std::invalid_argument("r has wrong dimension (it should be " + std::to_string(r.size()) + ")");
+    }
+    r = _r;
+  }
+  void set_Lx(const Eigen::VectorXd& _Lx) {
+    if (Lx.size() != _Lx.size()) {
+      throw std::invalid_argument("Lx has wrong dimension (it should be " + std::to_string(Lx.size()) + ")");
+    }
+    Lx = _Lx;
+  }
+  void set_Lu(const Eigen::VectorXd& _Lu) {
+    if (Lu.size() != _Lu.size()) {
+      throw std::invalid_argument("Lu has wrong dimension (it should be " + std::to_string(Lu.size()) + ")");
+    }
+    Lu = _Lu;
+  }
+  void set_Lxx(const Eigen::MatrixXd& _Lxx) {
+    if (Lxx.rows() != _Lxx.rows() && Lxx.cols() != _Lxx.cols()) {
```

Comment:
This statement is not correct. It should be an OR test.

Reply:
The same applies to other places.

Reply:
Thanks for the feedback! I fixed it, we'll see it in the next PR.